tests/unit/base_tests_zarrio.py

```diff
@@ -1123,108 +1113,102 @@ def test_soft_link_group(self):
         with ZarrIO(self.store[0], manager=get_foo_buildmanager(), mode='w') as write_io:
             write_io.write(foofile)
 
-        # with open(self.paths[0]+"/.zattrs", 'r') as f:
-        #     print(f.readlines())
-
         with ZarrIO(self.store[0], manager=get_foo_buildmanager(), mode='r') as read_io:
             with ZarrIO(self.store[1], mode='w') as export_io:
-                export_io.export(src_io=read_io,
-                                 write_args=dict(link_data=False))
+                export_io.export(src_io=read_io)
         with ZarrIO(self.store[1], manager=get_foo_buildmanager(), mode='r') as read_io:
             read_foofile2 = read_io.read()
-            # make sure the linked group is within the same file
-            self.assertEqual(read_foofile2.foo_link.container_source, self.store_path[1])
-            zarr_linkspec1 = zarr.open(self.store_path[0])['links'].attrs.asdict()['zarr_link'][0]
-            zarr_linkspec2 = zarr.open(self.store_path[1])['links'].attrs.asdict()['zarr_link'][0]
-            self.assertEqual(zarr_linkspec1.pop('source'), ".")
-            self.assertEqual(zarr_linkspec2.pop('source'), ".")
-            self.assertDictEqual(zarr_linkspec1, zarr_linkspec2)
-
-    def test_soft_link_dataset(self):
-        """Test that exporting a written file with soft linked datasets keeps links within the file."""
-        """Link to a dataset in the same file should have a link to the same new dataset in the new file """
-        pass  # TODO this test currently fails. It does not create a SoftLink in the original file.
-        """
-        foo1 = Foo('foo1', [1, 2, 3, 4, 5], "I am foo1", 17, 3.14)
-        foobucket = FooBucket('bucket1', [foo1])
-        foofile = FooFile(buckets=[foobucket], foofile_data=foo1.my_data)
-        with ZarrIO(self.store_paths[0], manager=get_foo_buildmanager(), mode='w') as write_io:
-            write_io.write(foofile, link_data=True)
-        print ("WRITE DONE")
-        with ZarrIO(self.store_paths[0], manager=get_foo_buildmanager(), mode='r') as read_io:
-            with ZarrIO(self.store_paths[1], mode='w') as export_io:
-                export_io.export(src_io=read_io, write_args=dict(link_data=False))
-        print(zarr.open(self.store_paths[0]).tree())
-        print(zarr.open(self.store_paths[1]).tree())
-        with ZarrIO(self.store_paths[1], manager=get_foo_buildmanager(), mode='r') as read_io:
-            read_foofile2 = read_io.read()
-            # make sure the linked dataset is within the same file
-            print(open(self.source_paths[1]+"/buckets/bucket1/foo_holder/foo1/.zattrs", 'r').read())
-            self.assertEqual(read_foofile2.foofile_data.path, self.source_paths[1])
-        """
+            if isinstance(self.store[1], str):
```

Comment:
When would self.store[1] not be a string? In setup(), it is `self.store = [f"file{i}.zarr" for i in range(3)]`

Reply:
Oh, I see. It's overridden elsewhere. That's a little confusing TBH, but I can suggest an improvement in a separate PR.